PiT 24.4: flow is looping trying to update types.d.ts'

### Description of the bug

starting certain hilla projects, flow tries to update 'types.d.ts'  but for some reason it does several times in the same run, and never ends.

```
java.lang.IllegalStateException: com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.base.devserver.AbstractDevServerRunner.getCause(AbstractDevServerRunner.java:772) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.getCause(AbstractDevServerRunner.java:768) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.handleRequestInternal(AbstractDevServerRunner.java:631) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.handleRequest(AbstractDevServerRunner.java:617) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1574) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:398) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.spring.SpringServlet.service(SpringServlet.java:106) ~[vaadin-spring-24.4.0.alpha7.jar:na]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.17.jar:6.0]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:642) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:408) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:313) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:277) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.servlet.mvc.ServletForwardingController.handleRequestInternal(ServletForwardingController.java:141) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:178) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:51) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:564) ~[tomcat-embed-core-10.1.17.jar:6.0]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.17.jar:6.0]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.1.3.jar:6.1.3]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.3.jar:6.1.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.1.3.jar:6.1.3]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.3.jar:6.1.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.1.3.jar:6.1.3]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.3.jar:6.1.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:340) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:391) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.updateIfContentMissing(TaskGenerateTsDefinitions.java:172) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.execute(TaskGenerateTsDefinitions.java:95) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:339) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:359) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:300) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[na:na]

2024-02-16T08:48:06.282+01:00 ERROR 1143 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[springServlet]        : Servlet.service() for servlet [springServlet] threw exception

com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.updateIfContentMissing(TaskGenerateTsDefinitions.java:172) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.execute(TaskGenerateTsDefinitions.java:95) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:339) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:359) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:300) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[na:na]

2024-02-16T08:48:06.283+01:00 ERROR 1143 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [com.vaadin.flow.server.ServiceException: java.lang.IllegalStateException: com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************


] with root cause

com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.updateIfContentMissing(TaskGenerateTsDefinitions.java:172) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.execute(TaskGenerateTsDefinitions.java:95) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:339) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:359) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:300) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[na:na]

2024-02-16T08:48:07.939+01:00 ERROR 1143 --- [nio-8080-exec-3] c.v.flow.server.DefaultErrorHandler      :

java.lang.IllegalStateException: com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.base.devserver.AbstractDevServerRunner.getCause(AbstractDevServerRunner.java:772) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.getCause(AbstractDevServerRunner.java:768) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.handleRequestInternal(AbstractDevServerRunner.java:631) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.handleRequest(AbstractDevServerRunner.java:617) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1574) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:398) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.spring.SpringServlet.service(SpringServlet.java:106) ~[vaadin-spring-24.4.0.alpha7.jar:na]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.17.jar:6.0]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:642) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:408) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:313) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:277) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.servlet.mvc.ServletForwardingController.handleRequestInternal(ServletForwardingController.java:141) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:178) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:51) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:564) ~[tomcat-embed-core-10.1.17.jar:6.0]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.1.3.jar:6.1.3]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.17.jar:6.0]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.1.3.jar:6.1.3]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.3.jar:6.1.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.1.3.jar:6.1.3]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.3.jar:6.1.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.1.3.jar:6.1.3]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.3.jar:6.1.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:340) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:391) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-10.1.17.jar:10.1.17]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.updateIfContentMissing(TaskGenerateTsDefinitions.java:172) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.execute(TaskGenerateTsDefinitions.java:95) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:339) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:359) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:300) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[na:na]

2024-02-16T08:48:07.941+01:00 ERROR 1143 --- [nio-8080-exec-3] o.a.c.c.C.[.[.[/].[springServlet]        : Servlet.service() for servlet [springServlet] threw exception

com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.updateIfContentMissing(TaskGenerateTsDefinitions.java:172) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.execute(TaskGenerateTsDefinitions.java:95) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:339) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:359) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:300) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[na:na]

2024-02-16T08:48:07.942+01:00 ERROR 1143 --- [nio-8080-exec-3] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [com.vaadin.flow.server.ServiceException: java.lang.IllegalStateException: com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************


] with root cause

com.vaadin.flow.server.ExecutionFailedException:

***************************************************************************
*  TypeScript type declaration file 'types.d.ts' has been updated to the  *
*  latest version by Vaadin. Previous content has been backed up on       *
*  'types.d.ts.flowBackup' file. Please verify that the updated           *
*  'types.d.ts' file contains configuration needed for your project, and  *
*  then delete the backup file.                                           *
***************************************************************************



	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.updateIfContentMissing(TaskGenerateTsDefinitions.java:172) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.TaskGenerateTsDefinitions.execute(TaskGenerateTsDefinitions.java:95) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:339) ~[flow-server-24.4.0.alpha7.jar:24.4.0.alpha7]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:359) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:300) ~[vaadin-dev-server-24.4.0.alpha7.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[na:na]


```

### Expected behavior

since previous file was already created by flow, file should be updated silently

### Minimal reproducible example

```
git clone https://github.com/vaadin/hilla-crm-tutorial.git
cd hilla-crm-tutorial
git checkout v24.4
mvn
```

### Versions

- Vaadin / Flow version: 24.4.0.alpha4

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java
@@ -72,7 +72,7 @@ public class TaskGenerateTsDefinitions extends AbstractTaskClientGenerator {
             """;
 
     static final String TS_DEFINITIONS = "types.d.ts";
-    static final Pattern COMMENT_LINE = Pattern.compile("(?m)//.*\\R");
+    static final Pattern COMMENT_LINE = Pattern.compile("(?m)^/[/*].*\\R");
     private final Options options;
 
     /**
@@ -131,11 +131,11 @@ public class TaskGenerateTsDefinitions extends AbstractTaskClientGenerator {
                         "Cannot read " + TS_DEFINITIONS + " contents", ex);
             }
 
-            String uncommentedDefaultContent = COMMENT_LINE
-                    .matcher(defaultContent).replaceAll("");
-            if (compareIgnoringEOL(content, defaultContent, String::equals)
-                    || compareIgnoringEOL(content, uncommentedDefaultContent,
-                            String::contains)) {
+            String uncommentedDefaultContent = removeComments(defaultContent);
+            if (compareIgnoringIndentationAndEOL(content, defaultContent,
+                    String::equals)
+                    || compareIgnoringIndentationAndEOL(content,
+                            uncommentedDefaultContent, String::contains)) {
                 log().debug("{} is up-to-date", TS_DEFINITIONS);
             } else if (content.contains(DECLARE_CSS_MODULE)) {
                 log().debug(
@@ -147,8 +147,8 @@ public class TaskGenerateTsDefinitions extends AbstractTaskClientGenerator {
                 log().debug(
                         "Updating custom {} to add '*.css?inline' module declaration",
                         TS_DEFINITIONS);
-                boolean customContent = hasCustomContent(content);
-                if (customContent) {
+                UpdateMode updateMode = computeUpdateMode(content);
+                if (updateMode == UpdateMode.UPDATE_AND_THROW) {
                     try {
                         Path backupFile = tsDefinitions.toPath().getParent()
                                 .resolve(TS_DEFINITIONS + ".flowBackup");
@@ -163,15 +163,17 @@ public class TaskGenerateTsDefinitions extends AbstractTaskClientGenerator {
                     }
                 }
                 try {
-                    if (customContent) {
-                        Files.writeString(tsDefinitions.toPath(),
-                                uncommentedDefaultContent,
-                                StandardOpenOption.APPEND);
-                        throw new ExecutionFailedException(UPDATE_MESSAGE);
-                    } else {
+                    if (updateMode == UpdateMode.REPLACE) {
                         Files.writeString(tsDefinitions.toPath(),
                                 defaultContent,
                                 StandardOpenOption.TRUNCATE_EXISTING);
+                    } else {
+                        Files.writeString(tsDefinitions.toPath(),
+                                uncommentedDefaultContent,
+                                StandardOpenOption.APPEND);
+                        if (updateMode == UpdateMode.UPDATE_AND_THROW) {
+                            throw new ExecutionFailedException(UPDATE_MESSAGE);
+                        }
                     }
                 } catch (IOException ex) {
                     throw new ExecutionFailedException(
@@ -182,24 +184,60 @@ public class TaskGenerateTsDefinitions extends AbstractTaskClientGenerator {
         }
     }
 
-    private boolean hasCustomContent(String content)
+    private enum UpdateMode {
+        REPLACE, UPDATE, UPDATE_AND_THROW
+    }
+
+    private UpdateMode computeUpdateMode(String content)
             throws ExecutionFailedException {
         try {
-            return !compareIgnoringEOL(content, getTemplateContent(".v1"),
-                    String::equals);
+            String templateContent = getTemplateContent(".v1");
+            if (compareIgnoringIndentationAndEOL(content, templateContent,
+                    String::equals)) {
+                // Current content has been written by Flow, can be replaced
+                return UpdateMode.REPLACE;
+            }
         } catch (IOException ex) {
             throw new ExecutionFailedException(
                     "Cannot read default " + TS_DEFINITIONS + ".v1 contents",
                     ex);
         }
+
+        try {
+            String templateContent = getTemplateContent(".hilla");
+            String uncommentedContent = removeComments(content);
+            if (compareIgnoringIndentationAndEOL(uncommentedContent,
+                    templateContent, String::equals)) {
+                // Current content is compatible with what we expect to be in a
+                // Hilla application. Flow contents can be appended silently.
+                return UpdateMode.UPDATE;
+            }
+        } catch (IOException ex) {
+            throw new ExecutionFailedException(
+                    "Cannot read default " + TS_DEFINITIONS + ".hilla contents",
+                    ex);
+        }
+        // types.d.ts has been customized, but does not seem to contain content
+        // required by flow. Append what is needed and throw an exception so
+        // that developer can check the changes are ok
+        return UpdateMode.UPDATE_AND_THROW;
     }
 
-    // Normalize EOL and removes potential EOL at the end of the FILE
-    private static boolean compareIgnoringEOL(String content1, String content2,
-            BiPredicate<String, String> compareFn) {
-        return compareFn.test(
-                content1.replace("\r\n", "\n").replaceFirst("\n$", ""),
-                content2.replace("\r\n", "\n").replaceFirst("\n$", ""));
+    private static String removeComments(String content) {
+        return COMMENT_LINE.matcher(content).replaceAll("");
+    }
+
+    private static boolean compareIgnoringIndentationAndEOL(String content1,
+            String content2, BiPredicate<String, String> compareFn) {
+        return compareFn.test(replaceIndentationAndEOL(content1),
+                replaceIndentationAndEOL(content2));
+    }
+
+    // Normalize EOL and removes indentation and potential EOL at the end of the
+    // FILE
+    static String replaceIndentationAndEOL(String text) {
+        return text.replace("\r\n", "\n").replaceFirst("\n$", "")
+                .replaceAll("(?m)^\\s+", "");
     }
 
     private String getTemplateContent(String suffix) throws IOException {

--- a/flow-server/src/main/resources/com/vaadin/flow/server/frontend/types.d.ts.hilla
+++ b/flow-server/src/main/resources/com/vaadin/flow/server/frontend/types.d.ts.hilla
@@ -1,0 +1,59 @@
+declare module '*.module.css' {
+  declare const styles: Record<string, string>;
+  export default styles;
+}
+declare module '*.module.sass' {
+  declare const styles: Record<string, string>;
+  export default styles;
+}
+declare module '*.module.scss' {
+  declare const styles: Record<string, string>;
+  export default styles;
+}
+declare module '*.module.less' {
+  declare const classes: Record<string, string>;
+  export default classes;
+}
+declare module '*.module.styl' {
+  declare const classes: Record<string, string>;
+  export default classes;
+}
+
+declare module '*.css';
+declare module '*.sass';
+declare module '*.scss';
+declare module '*.less';
+declare module '*.styl';
+
+declare module '*.svg' {
+  const ref: string;
+  export default ref;
+}
+declare module '*.bmp' {
+  const ref: string;
+  export default ref;
+}
+declare module '*.gif' {
+  const ref: string;
+  export default ref;
+}
+declare module '*.jpg' {
+  const ref: string;
+  export default ref;
+}
+declare module '*.jpeg' {
+  const ref: string;
+  export default ref;
+}
+declare module '*.png' {
+  const ref: string;
+  export default ref;
+}
+declare module '*.avif' {
+  const ref: string;
+  export default ref;
+}
+declare module '*.webp' {
+  const ref: string;
+  export default ref;
+}

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitionsTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitionsTest.java
@@ -133,6 +133,72 @@ public class TaskGenerateTsDefinitionsTest {
     }
 
     @Test
+    public void tsDefinition_oldHillaContents_tsDefinitionUpdated()
+            throws Exception {
+        Path typesTSfile = new File(outputFolder, "types.d.ts").toPath();
+        String hillaTsDef = IOUtils.toString(
+                getClass().getResourceAsStream(TS_DEFINITIONS + ".hilla"),
+                UTF_8);
+        Files.writeString(typesTSfile, hillaTsDef);
+        taskGenerateTsDefinitions.execute();
+        Assert.assertFalse(
+                "Should not generate types.d.ts when already existing",
+                taskGenerateTsDefinitions.shouldGenerate());
+        String updatedContent = Files.readString(typesTSfile);
+        MatcherAssert.assertThat("types.d.ts should have been updated",
+                updatedContent,
+                CoreMatchers.containsString(readExpectedContent(true)));
+        MatcherAssert.assertThat("types.d.ts should contain original content",
+                updatedContent, CoreMatchers.containsString(hillaTsDef));
+    }
+
+    @Test
+    public void tsDefinition_oldHillaContents_ignoringMultilineComments_tsDefinitionUpdated()
+            throws Exception {
+        Path typesTSfile = new File(outputFolder, "types.d.ts").toPath();
+        String hillaTsDef = IOUtils.toString(
+                getClass().getResourceAsStream(TS_DEFINITIONS + ".hilla"),
+                UTF_8);
+        // do not care about comment type (single or multi line)
+        hillaTsDef = hillaTsDef.replaceAll("(?m)^(\\s*declare.*)$",
+                "/* a comment */\n$1\n");
+        Files.writeString(typesTSfile, hillaTsDef);
+        taskGenerateTsDefinitions.execute();
+        Assert.assertFalse(
+                "Should not generate types.d.ts when already existing",
+                taskGenerateTsDefinitions.shouldGenerate());
+        String updatedContent = Files.readString(typesTSfile);
+        MatcherAssert.assertThat("types.d.ts should have been updated",
+                updatedContent,
+                CoreMatchers.containsString(readExpectedContent(true)));
+        MatcherAssert.assertThat("types.d.ts should contain original content",
+                updatedContent, CoreMatchers.containsString(hillaTsDef));
+    }
+
+    @Test
+    public void tsDefinition_oldHillaContents_ignoringSingleLineComments_tsDefinitionUpdated()
+            throws Exception {
+        Path typesTSfile = new File(outputFolder, "types.d.ts").toPath();
+        String hillaTsDef = IOUtils.toString(
+                getClass().getResourceAsStream(TS_DEFINITIONS + ".hilla"),
+                UTF_8);
+        // do not care about comment type (single or multi line)
+        hillaTsDef = hillaTsDef.replaceAll("(?m)^(\\s*declare.*)$",
+                "// a comment\n$1\n");
+        Files.writeString(typesTSfile, hillaTsDef);
+        taskGenerateTsDefinitions.execute();
+        Assert.assertFalse(
+                "Should not generate types.d.ts when already existing",
+                taskGenerateTsDefinitions.shouldGenerate());
+        String updatedContent = Files.readString(typesTSfile);
+        MatcherAssert.assertThat("types.d.ts should have been updated",
+                updatedContent,
+                CoreMatchers.containsString(readExpectedContent(true)));
+        MatcherAssert.assertThat("types.d.ts should contain original content",
+                updatedContent, CoreMatchers.containsString(hillaTsDef));
+    }
+
+    @Test
     public void customTsDefinition_missingFlowContents_tsDefinitionUpdatedAndExceptionThrown()
             throws Exception {
         Path typesTSfile = new File(outputFolder, "types.d.ts").toPath();
@@ -248,11 +314,11 @@ public class TaskGenerateTsDefinitionsTest {
                             [Ref in string]?: SchemaObject;
                         };
                         declare module '*.css?inline' {
-
+                          something
                           import type { CSSResultGroup } from 'lit';
-
+                          custom
                           const content: CSSResultGroup;
-
+                          added
                           export default content;
                         }
                         export declare const jtdForms: readonly ["elements", "values", "discriminator", "properties", "optionalProperties", "enum", "type", "ref"];

